lib/graphql/execution/interpreter/runtime.rb

```diff
@@ -332,20 +332,26 @@ def continue_field(path, value, field, type, ast_node, next_selections, is_non_n
             inner_type = type.of_type
             idx = 0
             scoped_context = context.scoped_context
-            value.each do |inner_value|
-              next_path = path.dup
-              next_path << idx
-              next_path.freeze
-              idx += 1
-              set_type_at_path(next_path, inner_type)
-              # This will update `response_list` with the lazy
-              after_lazy(inner_value, owner: inner_type, path: next_path, scoped_context: scoped_context, field: field, owner_object: owner_object, arguments: arguments) do |inner_inner_value|
-                continue_value = continue_value(next_path, inner_inner_value, field, inner_type.non_null?, ast_node)
-                if HALT != continue_value
-                  continue_field(next_path, continue_value, field, inner_type, ast_node, next_selections, false, owner_object, arguments)
+            begin
+              value.each do |inner_value|
+                next_path = path.dup
+                next_path << idx
+                next_path.freeze
+                idx += 1
+                set_type_at_path(next_path, inner_type)
+                # This will update `response_list` with the lazy
+                after_lazy(inner_value, owner: inner_type, path: next_path, scoped_context: scoped_context, field: field, owner_object: owner_object, arguments: arguments) do |inner_inner_value|
+                  continue_value = continue_value(next_path, inner_inner_value, field, inner_type.non_null?, ast_node)
+                  if HALT != continue_value
+                    continue_field(next_path, continue_value, field, inner_type, ast_node, next_selections, false, owner_object, arguments)
+                  end
                 end
               end
+            rescue NoMethodError
```

Comment:
@rmosolgo 👋 hope all is well.

I've ran into the situation where I'd get the `ListResultFailedError` in development and spend a lot of time trying to find out why it failed to build a list at the `SomeConnection.edges` field as the error says:

> Failed to build a GraphQL list result for field `ItemConnection.edges` at path `items.edges`.
Expected `[#<GraphQL::Pagination::Connection::Edge:0x00007fc1b7e07560 @connection=#<GraphQL::Pagination::ActiveRecordRelationConnection ...>>]` to implement `.each` to satisfy the GraphQL return type `[ItemEdge]`.

I couldn't figure out why the connection wasn't returning an array (due to the implement each method message) because I wasn't able to reproduce it when logging it in development. When I open the GraphQL gem with `bundle open graphql` and remove the rescue I get the original exception which helps me figure out what it meant exactly:

```
undefined method `organization' for nil:NilClass
/app/graphql/types/organization_type.rb:9:in `authorized?'
```

I'm not sure I understand #2588 completely for which this error was introduced for, but I'm wondering if this rescue is a bit too catch all? Curious if you have any opinions.

Reply:
> a bit too catch all

Ahh, I totally agree. Let me see about tightening up that rescue.